src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java

```diff
@@ -135,13 +144,13 @@ public void run() {
             
             CancerStudy cancerStudy = getCancerStudy(properties);
             
-            if (segmentDataExistsForCancerStudy(cancerStudy)) {
+            if (!updateMode && segmentDataExistsForCancerStudy(cancerStudy)) {
                  throw new IllegalArgumentException("Seg data for cancer study " + cancerStudy.getCancerStudyStableId() + " has already been imported: " + dataFile);
             }
         
             importCopyNumberSegmentFileMetadata(cancerStudy, properties);
             importCopyNumberSegmentFileData(cancerStudy, dataFile);
-            DaoCopyNumberSegment.createFractionGenomeAlteredClinicalData(cancerStudy.getInternalId());
+            DaoCopyNumberSegment.createFractionGenomeAlteredClinicalData(cancerStudy.getInternalId(), processedSampleIds, updateMode);
             if( MySQLbulkLoader.isBulkLoad()) {
```

Comment:
this if seems wrong as `importData` is always calling  `MySQLbulkLoader.flushAll();`

Reply:
I tried to improve it here https://github.com/cBioPortal/cbioportal-core/pull/47/commits/e3ec5d6064c9a7b3b0b19f19e28a674738ce4469